src/nvim/testdir/test_search.vim

```diff
@@ -0,0 +1,242 @@
+" Test for the search command
+
+func Test_search_cmdline()
+  if !exists('+incsearch')
+    return
+  endif
+  " need to disable char_avail,
+  " so that expansion of commandline works
+  call test_disable_char_avail(1)
```

Comment:
Can be tested with a screen test. See `test/functional/ui/inccommand_spec.lua` , 'inccommand' is quite similar :)